cloud/scope/powervs_cluster_test.go

```diff
@@ -1156,6 +1156,851 @@ func TestGetResourceGroupID(t *testing.T) {
+			IBMVPCClient: mockVpc,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							ID: nil,
+						},
+					},
+				},
+			},
+		}
+
+		mockVpc.EXPECT().GetLoadBalancerByName(gomock.Any()).Return(nil, nil)
+
+		requeue, err := clusterScope.ReconcileLoadBalancers()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When LoadBalancer ID is not set, the checkLoadBalancer function still returns a valid loadBalancerStatus, indicating that the load balancer exists in the cloud", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVpc,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							Name: "test-lb",
+							ID:   nil,
+						},
+					},
+				},
+			},
+		}
+
+		mockVpc.EXPECT().GetLoadBalancerByName(gomock.Any()).Return(&vpcv1.LoadBalancer{
+			ProvisioningStatus: ptr.To("active"),
+			Hostname:           ptr.To("test-lb-hostname"),
+			Name:               ptr.To("test-lb"),
+			ID:                 ptr.To("test-lb-instanceid"),
+		}, nil)
+
+		requeue, err := clusterScope.ReconcileLoadBalancers()
+		g.Expect(requeue).To(BeTrue())
+		g.Expect(err).To(BeNil())
+
+		loadBalancerStatus, ok := clusterScope.IBMPowerVSCluster.Status.LoadBalancers["test-lb"]
+		g.Expect(ok).To(BeTrue())
+		g.Expect(loadBalancerStatus.State).To(BeEquivalentTo(infrav1beta2.VPCLoadBalancerStateActive))
+		g.Expect(loadBalancerStatus.ID).To(Equal(ptr.To("test-lb-instanceid")))
+		g.Expect(loadBalancerStatus.Hostname).To(Equal(ptr.To("test-lb-hostname")))
+	})
+
+	t.Run("checkLoadBalancerPort should return an error when load balancer port matches with API server port", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterNetworkAPIServerPort := int32(9090)
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVpc,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+						{
+							Name: "test-lb",
+							ID:   nil,
+							AdditionalListeners: []infrav1beta2.AdditionalListenerSpec{
+								{
+									Port: 9090,
+								},
+							},
+						},
+					},
+				},
+			},
+			Cluster: &capiv1beta1.Cluster{
+				Spec: capiv1beta1.ClusterSpec{
+					ClusterNetwork: &capiv1beta1.ClusterNetwork{
+						APIServerPort: &clusterNetworkAPIServerPort,
+					},
+				},
+			},
+		}
+
+		mockVpc.EXPECT().GetLoadBalancerByName(gomock.Any()).Return(nil, nil)
+
+		requeue, err := clusterScope.ReconcileLoadBalancers()
+		g.Expect(requeue).To(BeFalse())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When createLoadBalancer returns error as resource group id is empty", func(*testing.T) {
```

Comment:
I see a couple of redundant test cases here and in `TestCreateLoadbalancer`, can we remove from one function?

Reply:
Removed redundant cases